cmd/diff.go

```diff
@@ -102,3 +105,99 @@ files.`,
   # referred to by './kubeconfig'.
   ks diff --kubeconfig=./kubeconfig -f ./pod.yaml`,
 }
+
+func initDiffCmd(cmd *cobra.Command, wd metadata.AbsPath, envSpec *envSpec, diffStrategy string) (kubecfg.DiffCmd, error) {
+	const (
+		remote = "remote"
+		local  = "local"
+	)
+
+	var err error
+
+	// ---------------------------------------------------------------------------
+	// Diff between expanded Kubernete objects and objects on a remote cluster
+	// ---------------------------------------------------------------------------
+	if envSpec.env2 == nil {
+		c := kubecfg.DiffRemoteCmd{}
+		c.DiffStrategy = diffStrategy
+		c.Client = &kubecfg.Client{}
+
+		c.Client.APIObjects, err = expandEnvCmdObjs(cmd, envSpec, wd)
```

Comment:
I'm not quite sure this is sufficient. If you do something like `ks diff local:default` we'll actually look for the environment called `local:default`, but I think we actually want to error out and report that `local:` isn't allowed in a `diff` with a single environment.